config/nrfconnect/app/sample-defaults.conf

```diff
@@ -118,3 +118,6 @@ CONFIG_MBEDTLS_CHACHAPOLY_C=n
 CONFIG_MBEDTLS_GCM_C=n
 CONFIG_MBEDTLS_RSA_C=n
 CONFIG_OBERON_MBEDTLS_AES_C=n
+
+# Enable Matter rotating device ID feature
+CONFIG_CHIP_ROTATING_DEVICE_ID=y
```

Comment:
Could we make the default false as in the other parts of the code?

Rationale being that this consumes some resources and spec-wise it is an optional feature.

Reply:
This config is by default set to false: https://github.com/project-chip/connectedhomeip/pull/13411/files#diff-bc10c06849de8756cfdadd674dbc3b8775bc75a337a054f962365defa05794daR111 for all Zephyr platforms. I enabled it only for nrfconnect examples, as we would like to support it in all our examples.

Reply:
Is the reason for that validating compilation?

I was thinking that in most cases, device  manufacturers are likely to first copy & paste some example code and then build upon this. In such cases, I was thinking to provide the smallest defaults and if changes are desired, they can be made afterwards.

Reply:
> Is the reason for that validating compilation?

Partially yes, obviously validating compilation in CI is something worth doing.

> I was thinking that in most cases, device manufacturers are likely to first copy & paste some example code and then build upon this. In such cases, I was thinking to provide the smallest defaults and if changes are desired, they can be made afterwards.

I think you are right. I had in my mind several cases that using rotating device ID could be useful, but in the end forcing user for disabling optional feature on his/her own doesn't seem to be right. I will disable it by default.